drivers/dp/swdp_bitbang.c

```diff
@@ -668,7 +668,11 @@ static int sw_port_off(const struct device *dev)
 	}
 
 	if (config->reset.port) {
+		#if defined(CONFIG_SWDP_MAINTAIN_RESET_DEASSERTED)
+		ret = gpio_pin_configure_dt(&config->reset, GPIO_OUTPUT_ACTIVE);
+		#else
 		ret = gpio_pin_configure_dt(&config->reset, GPIO_DISCONNECTED);
+		#endif
```

Comment:
C preprocessor directives shouldn't be indented, but you can even use the `IS_ENABLED` macro here.

```c
		ret = gpio_pin_configure_dt(&config->reset,
					    IS_ENABLED(CONFIG_SWDP_MAINTAIN_RESET_DEASSERTED)
						    ? GPIO_OUTPUT_ACTIVE
						    : GPIO_DISCONNECTED);
```